### PR TITLE
Clarify line breaking and indentation for theorems/definitions

### PR DESCRIPTION
**Pull request by Vierkantor**
Some clarifications in the style guide.

 * These rules apply for `def` and for `lemma`/`theorem`
 * Indent assumptions / conclusion after a line break
 * Short theorems can also fit on one line.

### CODE REVIEW

File: templates/contribute/style.md [hunk 1]
```diff
@@ -94,6 +94,8 @@ See our [documentation requirements](doc.html) for more suggestions and examples
 
 ### Structuring definitions and theorems ###
 
+These guidelines hold for declarations starting with `def`, `lemma` and `theorem`. For "theorem statement", also read "type declaration of a definition" and for "proof" also read "definition body".
```
Comment (Vierkantor) [inline templates/contribute/style.md:97]: Anyone know a good way to express "type declaration of a definition"?

File: templates/contribute/style.md [hunk 2]
```diff
@@ -107,6 +109,22 @@ theorem nat_case {P : nat → Prop} (n : nat) (H1: P 0) (H2 : ∀m, P (succ m))
...
+namespace nat
+
+lemma le_induction {P : ℕ → Prop} {m}
+  (h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
+  ∀ n, m ≤ n → P n :=
+by apply nat.less_than_or_equal.rec h0; exact h1
+
+def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
+  (hP : P n) : P m :=
+le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```
Comment (eric-wieser) [inline templates/contribute/style.md:123]: Playing devils advocate - is this more readable as
```suggestion
lemma le_induction {P : ℕ → Prop} {m}
(h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
  ∀ n, m ≤ n → P n :=
by apply nat.less_than_or_equal.rec h0; exact h1

def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
(hP : P n) :
  P m :=
le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```
or
```suggestion
lemma le_induction {P : ℕ → Prop} {m}
  (h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
    ∀ n, m ≤ n → P n :=
by apply nat.less_than_or_equal.rec h0; exact h1

def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
  (hP : P n) :
    P m :=
le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```
Essentially, do we want some subset of the addition rules:
* Wrapped arguments are not indented
* A newline must appear after the colon if the argument list spanned multiple lines
* If a newline appears after a colon, the theorem statement gains another level of indentation

Comment (Vierkantor) [reply templates/contribute/style.md:123]: Note that the goal of the pull request is descriptive: I wrote down what I understand as the current mathlib style. If we want to make changes to the conventions, let's make a poll on Zulip first.

For the record: I agree that it's almost always more readable to require a line break after `:` and `:=`, with the main exception being when the theorem is called  `bar_of_foo`, so we'd expect the conclusion line to be `foo -> bar`. If we turn `foo` into a named argument, I prefer keeping it on the same line. Something like:
```lean
lemma no_divisors_of_gcd_eq_one {x y : R}
  (h : gcd x y = 1) : ∀ z, z | x -> z | y -> is_unit z :=
sorry
```

File: templates/contribute/style.md (hunk 1 above)
Comment (robertylewis) [reply templates/contribute/style.md:97]: "type of a definition"? "declared type of a definition"?

File: templates/contribute/style.md [hunk 3]
```diff
@@ -94,6 +94,8 @@ See our [documentation requirements](doc.html) for more suggestions and examples
 
 ### Structuring definitions and theorems ###
```
Comment (robertylewis) [inline templates/contribute/style.md:95]:
```suggestion
### Structuring definitions and theorems
```

File: templates/contribute/style.md (hunk 1 above)
Comment (Vierkantor) [reply templates/contribute/style.md:97]:
```suggestion
These guidelines hold for declarations starting with `def`, `lemma` and `theorem`. For "theorem statement", also read "type of a definition" and for "proof" also read "definition body".
```

**Pull request merged**